pkg/plugin/plugin.go

```diff
@@ -16,17 +16,75 @@ package plugin
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strconv"
 
 	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/pkg/githubapp"
+	"github.com/google/go-github/v55/github"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-type GithubPlugin struct{}
+const (
+	// githubCategory is the justification category this plugin will be validating.
+	githubCategory               = "github"
+	respAnnotationKeyIssueOwner  = "github_issue_owner"
+	respAnnotationKeyIssueRepo   = "github_issue_repo"
+	respAnnotationKeyIssueNumber = "github_issue_number"
+)
 
-func (g *GithubPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
-	return nil, fmt.Errorf("Unimplemented")
+// issueMatcher is the mockable interface for the convenience of testing.
+type issueMatcher interface {
+	MatchIssue(ctx context.Context, issueURL string) (*pluginGitHubIssue, error)
+}
+
+// GithubPlugin is the implementation of jvspb.Validator interface.
+type GitHubPlugin struct {
+	validator issueMatcher
+}
+
+// NewGithubPlugin creates a new GithubPlugin.
+func NewGithubPlugin(ctx context.Context, ghClient *github.Client, ghApp *githubapp.GitHubApp) *GitHubPlugin {
+	return &GitHubPlugin{
+		validator: NewValidator(ghClient, ghApp),
+	}
 }
 
-func (g *GithubPlugin) GetUIData(ctx context.Context, req *jvspb.GetUIDataRequest) (*jvspb.UIData, error) {
+// Validate returns the validation result.
+func (g *GitHubPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
+	if got, want := req.Justification.Category, githubCategory; got != want {
+		return generateInvalidErrResq(fmt.Sprintf("failed to perform validation, expected category %q to be %q", got, want)), nil
+	}
+
+	info, err := g.validator.MatchIssue(ctx, req.Justification.Value)
```

Comment:
What if req.Justification.Value is empty, does it return invalidJustification error as we expected? maybe we should add a check on this specifically.

Reply:
Yes, if it's empty, it won't pass the regexp check, so I will return invalidJustification error.

Reply:
Ah, forgot about that, we are good here than, thanks!